### PR TITLE
Study listing brakes when no ref

### PR DESCRIPTION
**Pull request by antgonza**
Fixing that problem

### CODE REVIEW

**Comment by josenavas** [comment]
👍

**Comment by ElDeveloper** [comment]
Any chance a test can be added?

**Comment by antgonza** [comment]
I _think_ this is ready for review. These changes allows for:
- Fix study listing for new biom tables, like: deblur, human, uploaded
- Fix adding those bioms to analysis
- Fix displaying those bioms in the analysis page.
- Now the algorithm and parameters displayed for bioms are accurate and not hardcoded

Note that the analysis is still broken for non phylogenetic metrics, which should be another PR.

cc: @ElDeveloper, @josenavas

**Comment by antgonza** [comment]
Ahh, forgot to add that I'm also fixing a weird error between https/http in miniconda.

File: qiita_db/analysis.py [hunk 1]
```diff
@@ -857,20 +857,23 @@ def build_files(self,
 
             # figuring out if we are going to have duplicated samples, again
             # doing it here cause it's computational cheaper
-            # 1. merge samples per data_type, reference used and the command id
-            #    if not one of the tables is not 'Pick closed-reference OTUs'
-            #    it's safer to always rename. Note that grouped_samples is
-            #    basically how many biom tables we are going to create
+            # 1. merge samples per: data_type, reference used and
+            #                       the command id)
```
Comment (ElDeveloper) [inline qiita_db/analysis.py]: Looks like this line is out of place?

File: qiita_pet/templates/analysis_selected.html [hunk 2]
```diff
@@ -97,15 +96,23 @@
...
       <td>{{pid}} <a href="#" data-toggle="modal" data-target="#proc{{pid}}-settings-modal" onclick="return false;"><span class="glyphicon glyphicon-info-sign"></span></a></td>
       <td>{{proc_info[pid]["data_type"]}}</td>
       <td>{{proc_info[pid]["processed_date"]}}</td>
       <td>{{proc_info[pid]["algorithm"]}}</td>
-      <td>{{proc_info[pid]["reference_name"]}} {{proc_info[pid]["reference_version"]}}</td>
+      <td>
+        <small>
+          {% if proc_info[pid]["params"] is not None %}
+            {% for key, val in viewitems(proc_info[pid]["params"]) %}
+              <i>{{key}}:</i> {{val}} <br/>
```
Comment (ElDeveloper) [inline qiita_pet/templates/analysis_selected.html]: I think you want the `:` outside of the `</i>`, like in the section below.

File: qiita_pet/templates/list_studies.html [hunk 3]
```diff
@@ -67,10 +67,15 @@
...
+      var proc_data_table = '<h4>Processed Data</h4><table class="table" cellpadding="5" cellspacing="0" border="0" style="padding-left:50px;width:80%"><tr><th></th><th>ID</th><th>Data type</th><th>Processed Date</th><th>Samples</th><th>Algorithm</th><th>Parameters</th></tr>';
       for(i=0;i<d.proc_data_info.length;i++) {
         var proc_data = d.proc_data_info[i];
-        proc_data_table += '<tr><td><input type="button" class="btn btn-sm" value="Add" onclick="sel_proc_data(\''+ name +'\' ,'+ row +','+ proc_data.pid +')"></td><td>' + proc_data.pid + '</td><td>' + proc_data.data_type + '</td><td>' + proc_data.processed_date + '</td><td>' + proc_data.algorithm + '</td><td>' + proc_data.reference_name + ' ' + proc_data.reference_version + '</td><td>' + proc_data.samples.length + '</td></tr>';
+        var params = '';
+        var p = proc_data.params;
+        for (var key in p) {
+          params += '<i>' + key + ':</i> ' + p[key] + '<br/>';
```
Comment (ElDeveloper) [inline qiita_pet/templates/list_studies.html]: `:` should probably go outside? ... maybe 🤔

File: qiita_pet/test/test_analysis_handlers.py [hunk 4]
```diff
@@ -48,6 +48,17 @@ def test_post(self):
...
         self.assertEqual(response.code, 200)
 
+    def test_post_other_params(self):
+        post_args = {
+            'rarefaction-depth': '',
+            'merge-duplicated-sample-ids': 'on',
+            'commands': ['16S#command']
+        }
+        response = self.post('/analysis/wait/1', post_args)
+        print response.body
```
Comment (ElDeveloper) [inline qiita_pet/test/test_analysis_handlers.py]: Can you remove this line?

File: qiita_pet/test/test_analysis_handlers.py [hunk 5]
```diff
@@ -64,6 +75,14 @@ def test_get(self):
         self.assertEqual(response.code, 200)
 
 
+class TestSelectedSamplesHandler(TestHandlerBase):
+    def test_get(self):
+        response = self.get('/analysis/selected/')
+        # Make sure page response loaded sucessfully
+        print response.body
```
Comment (ElDeveloper) [inline qiita_pet/test/test_analysis_handlers.py]: Can you remove this line?

File: qiita_db/support_files/patches/41.sql [hunk 6]
```diff
@@ -0,0 +1,6 @@
+-- Oct 29, 2016
```
Comment (josenavas) [inline qiita_db/support_files/patches/41.sql]: Can you modify the HTML and dbs files to reflect this change? Thanks!

**Pull request merged**